source/extensions/common/wasm/context.cc

```diff
@@ -1364,6 +1391,13 @@ Http::FilterHeadersStatus Context::encode100ContinueHeaders(Http::ResponseHeader
 
 Http::FilterHeadersStatus Context::encodeHeaders(Http::ResponseHeaderMap& headers,
                                                  bool end_stream) {
+  if (!in_vm_context_created_) {
+    // If the request is invalid then onRequestHeaders() will not be called and neither will
+    // onCreate() then sendLocalReply be called which will call this function. In this case we
+    // need to call onCreate() so that the Context inside the VM is created before the
+    // onResponseHeaders() call.
+    onCreate(root_context_id_);
```

Comment:
I have an issue with this change, since it breaks normal HTTP request/response flow, and suddenly extensions will see a response to a request that it never received. If HTTP request is rejected by a lower layer (e.g. HTTP parser) or earlier extensions, and it never reaches our extension, then I don't see why HTTP response to that request should reach it.

As such, I think we should return `Http::FilterDataStatus::Continue`.

Reply:
SGTM.  This matches the Envoy native filter behavior, but I agree that is odd.  That said, it is a weird case.  The filters normally run for the response back to downstream, and now we are saying that it will not happen for certain responses (i.e. malformed headers). Should I create an issue in envoyproxy/envoy asking what the correct behavior should be in general (not just for Wasm) ?

Reply:
Done (code change)

Reply:
Yeah, please open an issue upstream.

Reply:
https://github.com/envoyproxy/envoy/issues/11447